### ISSUE TITLE
Cooldown broken?

### ISSUE THREAD

**Issue by BondiBear** [ccd0/4chan-x#456]
Since last upd to 1.11.8.8 cooldown seems broken. No countdown and just get 4chan warning I need to wait x seconds longer.

**Comment by ccd0** [comment]
Any error messages in your console (Control+Shift+J)?
Can you export and post your settings?
Double-checked that Cooldowns is enabled in your settings?
Also, if you're posting a thread via the original post form rather than the Quick Reply, or if you posted before updating, it is expected that you will not see a cooldown from that post.

**Comment by Davidj361** [comment]
I have the same issue.
[code]
Handler function threw an exception: [Exception... "Component returned failure code: 0x80004002 (NS_NOINTERFACE) [nsIWebProgress.DOMWindow]"  nsresult: "0x80004002 (NS_NOINTERFACE)"  location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js :: TabActor.prototype._docShellsToWindows/< :: line 1132"  data: no]
Stack: TabActor.prototype._docShellsToWindows/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js:1132:11
TabActor.prototype._docShellsToWindows@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js:1129:1
TabActor.prototype._notifyDocShellsUpdate@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js:1155:19
TabActor.prototype._onDocShellCreated/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js:1107:7
makeInfallible/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/DevToolsUtils.js:82:14
Line: 1132, column: 0
[/code]

**Comment by ccd0** [comment]
@Davidj361
That error doesn't look related to 4chan X.  Are there any errors at the time you make a post, when it should be starting the cooldown timer?  Make sure you look in the browser console (Control+Shift+J), not the web console.

**Comment by Davidj361** [comment]
It seems to fix itself when i refresh the page. Here's some errors though.
```
TypeError: post is undefined 4chan-X.user.js:9026:0
TypeError: can't access dead object 4chan-X.user.js:14213:0
TypeError: post is undefined 4chan-X.user.js:9026:0
```

**Comment by ccd0** [comment]
>TypeError: post is undefined 4chan-X.user.js:9026:0

Do you have the stack trace for this?

**Comment by Davidj361** [comment]
How do I get a stack trace?

**Comment by ccd0** [comment]
Never mind, for some reason for a moment I thought it showed stack traces by default, but it doesn't.

That said, if you could do something like change

    categorize: function(post) {
      if (post.thread === 'new') {
        return {
          type: 'thread'
        };
      } else {
        return {
          type: !!post.file ? 'image' : 'reply',
          threadID: +post.thread
        };
      }
    },

to

    categorize: function(post) {
     try {
      if (post.thread === 'new') {
        return {
          type: 'thread'
        };
      } else {
        return {
          type: !!post.file ? 'image' : 'reply',
          threadID: +post.thread
        };
      }
     } catch(err) {
       c.log(err.stack);
       throw err;
     }
    },

that would help.

**Comment by ccd0** [comment]
Also, what browser and browser version are you using?

**Comment by ccd0** [comment]
Another quick thing to check:  If you type `Date.now()` in the Javascript console, what does it give you?  Trying to make sure it's accurate.

**Comment by Davidj361** [comment]
Firefox 40.0.2
```
Date.now()
1439771115848
```

**Comment by ccd0** [comment]
@Davidj361
Can you make the bug happen again, or does it seem to be gone?

**Comment by Davidj361** [comment]
It seems to be gone, but I'll reply if it happens again.

**Comment by ccd0** [comment]
Alright, I think I've found the problem.  It occurs when you make a post in one tab and then switch to another, already open tab in which you have not yet opened the quick reply.  I'll push a fix shortly.

**Comment by Davidj361** [comment]
Thank you.

**Comment by ccd0** [comment]
Alright, this should be fixed by 6642e6e1cf6e1cc3f9efc4b5cf76e62f56328482 in v1.11.8.9 and 
v1.11.9.2.

**Comment by BondiBear** [comment]
Seems fixed! Thank you!
